llvm/runtimes/CMakeLists.txt

```diff
@@ -138,37 +142,43 @@ endfunction()
 # before the just-built compiler can pass the configuration tests.
 get_compiler_rt_path(compiler_rt_path)
 if(compiler_rt_path)
+  # If the user did not specify the targets infer them from the runtimes.
   if(NOT LLVM_BUILTIN_TARGETS)
+    if("compiler-rt" IN_LIST LLVM_ENABLE_RUNTIMES)
+      list(APPEND LLVM_BUILTIN_TARGETS "default")
+    endif()
+    foreach(name ${LLVM_RUNTIME_TARGETS})
+      if("compiler-rt" IN_LIST RUNTIMES_${name}_LLVM_ENABLE_RUNTIMES)
+        list(APPEND LLVM_BUILTIN_TARGETS ${name})
+      endif()
+    endforeach()
+  endif()
+  if("default" IN_LIST LLVM_BUILTIN_TARGETS)
     builtin_default_target(${compiler_rt_path}
       DEPENDS clang-resource-headers)
+    list(REMOVE_ITEM LLVM_BUILTIN_TARGETS "default")
```

Comment:
What's the reason for doing this? We shouldn't be modifying global variables controlled by the user.

Reply:
That's what the old code does. I could put it in a temporary variable instead if that would be better.